lib/cc/analyzer/engines_config_builder.rb

```diff
@@ -29,12 +29,22 @@ def run
 
       private
 
-      attr_reader :include_paths
+      attr_reader :include_paths_builder
+
+      def include_paths(engine_excludes)
+        if @requested_paths.empty?
+          IncludePathsBuilder.new(global_excludes).build(engine_excludes)
+        else
+          IncludePathsBuilder.new([], @requested_paths).build
```

Comment:
Oops! This was left over from when we were discussing it and then I got sidetracked. Adding it back in.